client/assets/js/board.js

```diff
@@ -84,4 +86,33 @@ export default class Board {
 		});
 	}
 
+    checkCollisionWithYourself() {
+		//let snake = this.snakes[0];
+
+        this.snakes.forEach((snake, i) => {
+
+            let firstBodyPart = snake.bodyParts[0];
+            snake.bodyParts.forEach((bodyPart, index) => {
+                if (index === 0) {
+                    return;
+                }
+
+                if (firstBodyPart.x < bodyPart.x + bodyPart.width &&
+                    firstBodyPart.x + firstBodyPart.width > bodyPart.x &&
+                    firstBodyPart.y < bodyPart.y + bodyPart.height &&
+                    firstBodyPart.height + firstBodyPart.y > bodyPart.y) {
+
+                	this.removeSnakeFromArray(i);
+                }
+            });
+		})
+    }
+
+    removeSnakeFromArray(i) {
+		this.snakes[i].bodyParts.forEach((part, index) => {
```

Comment:
It's weird that the board manipulates directly the body parts. Maybe a `Snake#remove` (or `delete` or `destroy`) method would be more appropriate.